sdk/data/aztables/internal/auth/challenge_policy.go

```diff
@@ -0,0 +1,236 @@
+	tenant = strings.ReplaceAll(tenant, ",", "")
+	return &tenant
+}
+
+type challengePolicyError struct {
+	err error
+}
+
+func (c *challengePolicyError) Error() string {
+	return c.err.Error()
+}
+
+func (*challengePolicyError) NonRetriable() {
+	// marker method
+}
+
+func (c *challengePolicyError) Unwrap() error {
+	return c.err
+}
+
+var _ errorinfo.NonRetriable = (*challengePolicyError)(nil)
+
+// sets the scp.scope and scp.tenantID from the WWW-Authenticate header
+func (scp *TablesChallengePolicy) findScopeAndTenant(resp *http.Response) error {
+	authHeader := resp.Header.Get("WWW-Authenticate")
+	if authHeader == "" {
+		return &challengePolicyError{err: errors.New("response has no WWW-Authenticate header for challenge authentication")}
+	}
+
+	// Strip down to auth and resource
+	// Format is "Bearer authorization=<site> resource=<site>" OR
+	// "Bearer authorization=<site> scope=<site> resource=<resource>"
+	authHeader = strings.ReplaceAll(authHeader, "Bearer ", "")
+
+	parts := strings.Split(authHeader, " ")
+
+	vals := map[string]string{}
+	for _, part := range parts {
+		subParts := strings.Split(part, "=")
+		if len(subParts) == 2 {
+			stripped := strings.ReplaceAll(subParts[1], "\"", "")
+			stripped = strings.TrimSuffix(stripped, ",")
+			vals[subParts[0]] = stripped
+		}
+	}
+
+	scp.tenantID = parseTenant(vals["authorization_uri"])
+	if scope, ok := vals["scope"]; ok {
+		scp.scope = &scope
+	} else if resource, ok := vals["resource_id"]; ok {
+		if !strings.HasSuffix(resource, "/.default") {
+			resource += "/.default"
+		}
+		scp.scope = &resource
+	} else {
+		return &challengePolicyError{err: errors.New("could not find a valid resource in the WWW-Authenticate header")}
+	}
+
+	return nil
+}
+
+func (k TablesChallengePolicy) getChallengeRequest(orig policy.Request) (*policy.Request, error) {
+	req, err := runtime.NewRequest(orig.Raw().Context(), "GET", orig.Raw().URL.String())
+	if err != nil {
+		return nil, &challengePolicyError{err: err}
+	}
+
+	req.Raw().Header = orig.Raw().Header
+	req.Raw().Header.Set("Content-Length", "0")
+	req.Raw().ContentLength = 0
+
+	copied := orig.Clone(orig.Raw().Context())
+	copied.Raw().Body = req.Body()
+	copied.Raw().ContentLength = 0
+	copied.Raw().Header.Set("Content-Length", "0")
+	err = copied.SetBody(streaming.NopCloser(bytes.NewReader([]byte{})), "application/json")
+	if err != nil {
+		return nil, &challengePolicyError{err: err}
+	}
+	copied.Raw().Header.Del("Content-Type")
+
+	return copied, err
+}
+
+type acquiringResourceState struct {
+	req *policy.Request
+	p   *TablesChallengePolicy
+}
+
+// acquire acquires or updates the resource; only one
+// thread/goroutine at a time ever calls this function
+func acquire(state interface{}) (newResource interface{}, newExpiration time.Time, err error) {
+	s := state.(acquiringResourceState)
+	tk, err := s.p.cred.GetToken(
+		s.req.Raw().Context(),
+		policy.TokenRequestOptions{
+			Scopes:   []string{*s.p.scope},
+			TenantID: *s.p.scope,
```

Comment:
This won't work with `azidentity` credentials until we address #14932. I thought we intended to remove this field to avoid suggesting it might have an effect; @jhendrixMSFT do you recall discussing that?

Reply:
If that's the case, wouldn't this fail for keyvault which uses the same logic? 
Reference: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/keyvault/internal/challenge_policy.go#L223-L236

Reply:
Only when the resource isn't in the credential's configured tenant. You would need to ensure it isn't in order to test multitenancy (the live test you're adding here doesn't).

Reply:
@chlowell we use it today in the implementation for ARM's multitenant auth [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/runtime/policy_bearer_token.go#L29).